core/auth/src/main/java/software/amazon/awssdk/auth/signer/AwsSignerExecutionAttribute.java

```diff
@@ -15,69 +15,463 @@
+            // Let's assume they're setting the region scope so that they can call the signer directly. If that's true, then it
+            // doesn't really matter what we store other than the region scope.
+            return new SelectedAuthScheme<>(CompletableFuture.completedFuture(new UnsetIdentity()),
+                                            new UnsetHttpSigner(),
+                                            AuthSchemeOption.builder()
+                                                            .schemeId("unset")
+                                                            .putSignerProperty(AwsV4aHttpSigner.REGION_NAME,
+                                                                               regionScopeString)
+                                                            .build());
+        }
+
+        return new SelectedAuthScheme<>(authScheme.identity(),
+                                        authScheme.signer(),
+                                        authScheme.authSchemeOption().copy(o -> o.putSignerProperty(AwsV4aHttpSigner.REGION_NAME,
+                                                                                                    regionScopeString)));
+    }
+
+    private static String serviceSigningNameReadMapping(SelectedAuthScheme<?> authScheme) {
+        if (authScheme == null) {
+            return null;
+        }
+        return authScheme.authSchemeOption().signerProperty(AwsV4FamilyHttpSigner.SERVICE_SIGNING_NAME);
+    }
+
+    private static <T extends Identity> SelectedAuthScheme<?> serviceSigningNameWriteMapping(SelectedAuthScheme<T> authScheme,
+                                                                                             String signingName) {
+        if (authScheme == null) {
+            // This is an unusual use-case.
+            // Let's assume they're setting the signing name so that they can call the signer directly. If that's true, then it
+            // doesn't really matter what we store other than the signing name.
+            return new SelectedAuthScheme<>(CompletableFuture.completedFuture(new UnsetIdentity()),
+                                            new UnsetHttpSigner(),
+                                            AuthSchemeOption.builder()
+                                                            .schemeId("unset")
+                                                            .putSignerProperty(AwsV4FamilyHttpSigner.SERVICE_SIGNING_NAME,
+                                                                               signingName)
+                                                            .build());
+        }
+
+        return new SelectedAuthScheme<>(authScheme.identity(),
+                                        authScheme.signer(),
+                                        authScheme.authSchemeOption()
+                                                  .copy(o -> o.putSignerProperty(AwsV4FamilyHttpSigner.SERVICE_SIGNING_NAME,
```

Comment:
It's not easy. The auth scheme option can change later on when auth option resolution happens, so we'd then lose any "writes" that happen before the auth option is resolved if we didn't store them anywhere. Engineering a workaround seems like a lot of complexity, so I'm not inclined.

Reply:
ah, my frame of mind was that anyone writing these "knows"/"expects" the corresponding scheme (since we currently only really have one scheme per service, it could be viewed that way, was my thinking), so it might lead to mismatched expectations. but I agree, if I frame it as, the writers are setting properties up prospectively and whether the corresponding scheme is eventually used or not, doesn't matter. I think the main thing that bothers me is that this is hanging off *Selected*AuthScheme, but I agree the complexity of engineering something else is not worth it.